File: prayer_app_flutter/lib/src/components/next_prayer_card.dart
import 'package:flutter/material.dart';
import 'package:material_design_icons_flutter/material_design_icons_flutter.dart';
import '../theme/app_theme.dart';
import '../providers/theme_provider.dart';

class NextPrayerCard extends StatelessWidget {
  final String name;
  final String countdown;
  final String adhanTime;

  const NextPrayerCard({
    super.key,
    required this.name,
    required this.countdown,
    required this.adhanTime,
  });

  @override
  Widget build(BuildContext context) {
    final tc = ThemeScope.of(context).current;
    return Container(
      constraints: const BoxConstraints(minHeight: SalahLayout.heroMinHeight),
      padding: const EdgeInsets.all(SalahLayout.heroPadding),
      decoration: BoxDecoration(
        color: tc.card,
        borderRadius: BorderRadius.circular(SalahLayout.heroRadius),
        border: Border.all(
          color: tc.accent.withValues(alpha: SalahLayout.heroBorderOpacity),
          width: SalahLayout.heroBorderWidth,
        ),
      ),
      child: Row(
        children: [
          // Icon box
          Container(
            width: SalahLayout.heroIconBoxSize,
            height: SalahLayout.heroIconBoxSize,
            decoration: BoxDecoration(
              color: tc.accent,
              borderRadius: BorderRadius.circular(SalahLayout.heroIconBoxRadius),
            ),
            child: Icon(
              MdiIcons.mosque,
              color: tc.backgroundStart,
              size: SalahLayout.heroIconSize,
            ),
          ),
          const SizedBox(width: SalahLayout.heroIconTextGap),
          // Text column
          Expanded(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                Text(
                  'Next Prayer: ${name.toUpperCase()}',
                  style: AppTypography.body(tc).copyWith(
                    fontSize: SalahLayout.heroLine1Size,
                    fontWeight: FontWeight.w500,
                    color: tc.textPrimary,
                  ),
                ),
                const SizedBox(height: 2),
                Row(
                  crossAxisAlignment: CrossAxisAlignment.baseline,
                  textBaseline: TextBaseline.alphabetic,
                  children: [
                    Text(
                      'Starts in ',
                      style: AppTypography.caption(tc).copyWith(
                        fontSize: SalahLayout.heroLine1Size,
                        color: tc.textMuted,
                      ),
                    ),
                    Text(
                      countdown,
                      style: AppTypography.titleLarge(tc).copyWith(
                        fontSize: SalahLayout.heroCountdownSize,
                        fontWeight: FontWeight.w700,
                        color: tc.textPrimary,
                      ),
                    ),
                  ],
                ),
                const SizedBox(height: 2),
                Text(
                  'Adhan at $adhanTime',
                  style: AppTypography.caption(tc).copyWith(
                    fontSize: SalahLayout.heroLine3Size,
                    color: tc.textMuted,
                  ),
                ),
              ],
            ),
          ),
        ],
      ),
    );
  }
}

File: prayer_app_expo/src/components/NextPrayerCard.js
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { Colors, Spacing, SalahLayout, Typography, interFont } from '../theme/theme';

export default function NextPrayerCard({ name, countdown, adhanTime }) {
    return (
        <View style={styles.card}>
            {/* Icon box */}
            <View style={styles.iconBox}>
                <MaterialCommunityIcons
                    name="mosque"
                    size={SalahLayout.heroIconSize}
                    color={Colors.backgroundStart}
                />
            </View>
            <View style={styles.gap} />
            {/* Text column */}
            <View style={styles.textCol}>
                <Text style={styles.line1}>
                    Next Prayer: {name.toUpperCase()}
                </Text>
                <View style={styles.countdownRow}>
                    <Text style={styles.startsIn}>Starts in </Text>
                    <Text style={styles.countdown}>{countdown}</Text>
                </View>
                <Text style={styles.line3}>Adhan at {adhanTime}</Text>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    card: {
        flexDirection: 'row',
        alignItems: 'center',
        alignSelf: 'stretch',
        minHeight: SalahLayout.heroMinHeight,
        padding: SalahLayout.heroPadding,
        backgroundColor: Colors.card,
        borderRadius: SalahLayout.heroRadius,
        borderWidth: SalahLayout.heroBorderWidth,
        borderColor: `rgba(212, 168, 71, ${SalahLayout.heroBorderOpacity})`,
    },
    iconBox: {
        width: SalahLayout.heroIconBoxSize,
        height: SalahLayout.heroIconBoxSize,
        borderRadius: SalahLayout.heroIconBoxRadius,
        backgroundColor: Colors.accentGold,
        justifyContent: 'center',
        alignItems: 'center',
    },
    gap: {
        width: SalahLayout.heroIconTextGap,
    },
    textCol: {
        flex: 1,
    },
    line1: {
        fontFamily: interFont('500'),
        fontSize: SalahLayout.heroLine1Size,
        color: Colors.textPrimary,
    },
    countdownRow: {
        flexDirection: 'row',
        alignItems: 'baseline',
        marginTop: 2,
    },
    startsIn: {
        fontFamily: interFont('400'),
        fontSize: SalahLayout.heroLine1Size,
        color: Colors.textMuted,
    },
    countdown: {
        fontFamily: interFont('700'),
        fontSize: SalahLayout.heroCountdownSize,
        color: Colors.textPrimary,
    },
    line3: {
        fontFamily: interFont('400'),
        fontSize: SalahLayout.heroLine3Size,
        color: Colors.textMuted,
        marginTop: 2,
    },
});
